src/status_im/transport/inbox.cljs

```diff
@@ -219,18 +221,22 @@
  :inbox/request-messages
  (fn [{:keys [db now]} [_ {:keys [from topics discover?]}]]
    (let [web3       (:web3 db)
+         account    (:account/account db)
          wnode      (get-current-wnode-address db)
-         topics     (or topics
-                        (map #(:topic %) (vals (:transport/chats db))))
-         from       (or from (:inbox/last-request db) nil)
-         sym-key-id (:inbox/sym-key-id db)]
-     {::request-messages {:wnode      wnode
-                          :topics     (if discover?
-                                        (conj topics (transport.utils/get-topic constants/contact-discovery))
-                                        topics)
-                          ;;TODO (yenda) fix from, right now mailserver is dropping us
-                          ;;when we send a requestMessage with a from field
-                          ;;:from       from
-                          :sym-key-id sym-key-id
-                          :web3       web3}
-      :db (assoc db :inbox/last-request (quot now 1000))})))
+         topics     (cond-> (or topics
+                                (map #(:topic %) (vals (:transport/chats db))))
+                      discover? (conj (transport.utils/get-topic constants/contact-discovery)))
+         from       (or from (:last-request account))
+         sym-key-id (:inbox/sym-key-id db)
+         now-in-s   (quot now 1000)
+         fx         {:data-store/save-account (assoc account :last-request now-in-s)
```

Comment:
It seems here we store  `:last-request` even if we:

1) Don't make the request because it comes in less than 10 seconds before the last
2) The request to the mailserver fails

In both cases we might lose some messages.

Seems like 1 it's easy to change, unless there are some other reason why we do that?

2 Is a bit harder, we should probably either store in only on success (which brings some complications), or if we want to store it optimistically like this then probably worth retrying the request a few times, give up and move on. 
Maybe not necessary in this PR but worth leaving a comment as something we need to address.